Update to work with Bundler & Rails 3.0 and make easier to apply

matellis opened the pull request:
Hello

Thanks for developing resque-statsd. We had some difficulty using this with Rails 3.0 & Bundler. Specifically, the way we had to use environment variables meant we couldn't simply add `extend Resque::Plugins::Statsd` in the right places like we do with other extensions because the initializer hadn't run yet.

Our patch doesn't work with Rails 2.x, it would need to be adjusted to try both `ENV['RAILS_ENV']` as well as Rails.env.

Thanks

Mat & Jon

MrJoy committed:
Changing to lazy-instantiation model, so we can switch to a more robust configuration model.

MrJoy committed:
Adding the gemspec...

MrJoy committed:
Hoisting defaults, adding env-search, fixing stupid.

MrJoy committed:
Making config actually work properly.

Using a separate helper class to avoid weirdness with module lifecycle (yes yes, I need to learn how that aspect of Ruby works in more depth.).

Commit:
Updated readme

jamster commented:
Thanks for contributing!  I've actually lost my statsd/graphite setup due to a cloud server failure and my lack of time to get it back up, so I haven't  messed around with it for quite some time.  But, you've given me incentive to get it back up and test all this.  Super busy this week, hope to get it in this weekend and I'll happily merge your commits.

Cheers,
Jason

matellis commented:
We have a working puppet recipe if that helps, let me know if you'd like a copy.

M.

On Aug 25, 2011, at 1:46 PM, jamster wrote:

> Thanks for contributing!  I've actually lost my statsd/graphite setup due to a cloud server failure and my lack of time to get it back up, so I haven't  messed around with it for quite some time.  But, you've given me incentive to get it back up and test all this.  Super busy this week, hope to get it in this weekend and I'll happily merge your commits.
> 
> Cheers,
> Jason
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jamster/resque-statsd/pull/1#issuecomment-1903774

jamster commented:
Yep, if you have that script handy, that would be great!

matellis commented:
Install puppet, then put this somewhere and run:

puppet apply graphite.pp

Enjoy!

M.
PS: Designed for ubuntu

On Aug 26, 2011, at 6:29 AM, jamster wrote:

> Yep, if you have that script handy, that would be great!
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jamster/resque-statsd/pull/1#issuecomment-1912542

jamster commented:
didn't see an attachment.  was there one?  (never used puppet before, is it like ruby gems where it will pull from a centralized source?)

matellis commented:
Here are the contents of the attachment. If you've not used puppet before it might be a bit of a pain. 

On an ubuntu system copy the contents of this file to graphite.pp and do the following:

gem install puppet
puppet apply graphite.pp

It should work but it might not, if your ubuntu system is setup in a different way you might find a few dependencies missing but it might just work. Add --debug to the puppet command if you want to debug it, or you can just read the recipe and run the commands/install the packages manually.

Good luck.

M.

On Aug 28, 2011, at 9:24 AM, jamster wrote:

> didn't see an attachment.  was there one?  (never used puppet before, is it like ruby gems where it will pull from a centralized source?)
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jamster/resque-statsd/pull/1#issuecomment-1923983

noahhl committed:
Add a before_perform_statsd method

noahhl committed:
Add a before_perform_statsd method

(Updated to match new queue/job-name split, etc.)

Conflicts:

	lib/resque/plugins/statsd.rb

MrJoy committed:
Namespace things a bit.

(So if you're tracking something BESIDES Resque jobs in your app, you don't get frustrated...)

noahhl committed:
Add a gemspec

Commit:
Logging split by job as well as queue in statsd

Commit:
resque is the default namespace, so removing from module

Commit:
Adding some monkey patching to monitor time spent in queue.

Commit:
updating instructions to include require in initializer

Commit:
typo

Commit:
Making this monkey patch work with a similar one adding functionality to the same Resque.push function
Old code was getting over-written by the last-included module monkey-patch

MrJoy committed:
Updating dependencies.

MrJoy committed:
Tidying up whitespace cruft.

MrJoy committed:
Merge with '--strategy ours' to hide commits on GH fork view.

MrJoy committed:
Tabs to spaces.

MrJoy committed:
Swallowing errors so statsd issues don't take down your app.

MrJoy committed:
Quarantine ourselves.

MrJoy committed:
Letting dependencies float, and removing explicit bundler dep as that's just silly.

MrJoy committed:
Updating dependencies.

MrJoy committed:
Freshening gemspec via new Jeweler build.

MrJoy committed:
Bumping version to reflect all the shiny little tidy-ups happening here.

MrJoy committed:
Updating authorship so people using our fork know who to kvetch at.

MrJoy committed:
Tidying up, and skipping a few files that don't belong in the gem.

MrJoy committed:
While we're at it, what is the point of including the test suite in the gem?

It just bloats deployments and doesn't actually make it easy/feasible to run the
tests out of the gems anyway.

So, if you wanna run the tests (if/when they ever exist) -- clone the repo.

MrJoy committed:
Updating gemspec.
